state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -957,8 +1025,115 @@ pub mod pallet {
 					debug_assert!(false, "Swap is not completed yet!");
 				}
 			}
+		}
 
-			Ok(())
+		// Returns swaps that have not expired yet and should be scheduled for retry.
+		fn refund_any_expired_swaps(
+			swaps: Vec<Swap>,
+			current_block: BlockNumberFor<T>,
+		) -> Vec<Swap> {
+			let (swaps_to_retry, swaps_to_refund): (Vec<Swap>, Vec<Swap>) =
+				swaps.into_iter().partition(|swap| {
+					swap.refund_params
+						.as_ref()
+						.map_or(true, |params| current_block < params.refund_if_reaches.into())
+				});
+
+			for swap in swaps_to_refund {
+				let destination_address = swap.refund_params.unwrap().refund_address;
+				match T::EgressHandler::schedule_egress(
+					swap.from,
+					swap.input_amount,
+					destination_address,
+					None,
+				) {
+					Ok(ScheduledEgressDetails { egress_id, egress_amount, fee_withheld }) => {
+						Self::deposit_event(Event::<T>::RefundEgressScheduled {
+							swap_id: swap.swap_id,
+							egress_id,
+							amount: egress_amount,
+							egress_fee: fee_withheld,
+						});
+					},
+					Err(err) => {
+						Self::deposit_event(Event::<T>::RefundEgressIgnored {
+							swap_id: swap.swap_id,
+							asset: swap.from,
+							amount: swap.input_amount,
+							reason: err.into(),
+						});
+					},
+				}
+			}
+
+			swaps_to_retry
+		}
+
+		fn process_swaps_for_block(
+			current_block: BlockNumberFor<T>,
+			block_to_process: BlockNumberFor<T>,
+		) -> Result<(), ()> {
+			let mut swaps_to_execute = SwapQueue::<T>::take(block_to_process);
+
+			loop {
+				if swaps_to_execute.is_empty() {
+					return Ok(())
+				}
+
+				match Self::execute_batch(swaps_to_execute.clone()) {
+					Ok(successful_swaps) => {
+						Self::process_swap_outcomes(&successful_swaps);
+						// Nothing to do here, all swaps are processed for block
+						return Ok(())
+					},
+					Err(err) => {
+						let (successful_swaps, failed_swaps) = {
```

Comment:
> since that is actually what they are

They are that, but they are also "successful" in a way (we only retry them because we some other swaps in the batch fail). I agree that this could be confusing though, but `next_batch` could aslo imply that it is a brand new one and has nothing to do with the previous batch, rather than being its subset. I renamed to `satisfactory_swaps` and added a comment.

Reply:
Mmm but they're only "satisfactory" in that batch. They might not be in another batch where the price is moved differently to the swaps in the new batch - I think `satisfactory_swaps` is better than `successful_swaps` but I still think `next_batch` is just true, and we don't have to make a judgement on how good or bad the swaps are (which changes depending on which batch their in).... i do think it's better now though, so ok to merge this as is